### ISSUE TITLE
Code catches wrong JSON error

### ISSUE THREAD

**Issue by jpmoral-factory** [uptick/pymyob#59]
Code catches `json.JSONDecodeError` but actual error raised is `simplejson.errors.JSONDecodeError`.

@jarekwg sorry about that mate, will open a PR soon.

**Comment by jpmoral-factory** [comment]
I don't think this is as simple as it first seemed. Different installs may be using different JSON libraries.

**Comment by jarekwg** [comment]
Bleh -- you just sent me down a massive rabbithole of investigation, exploring python history.

The inconsistency comes as a result of the `requests` library preferring to use `simplejson` if it is available in installed modules. It does this for legacy reasons, because `json` was only introduced into the standard library in python 2.6, and it was quite unperformant back then.
They've added a blurb to the requests package [explaining this behaviour](https://github.com/psf/requests/blob/c45a4dfe6bfc6017d4ea7e9f051d6cc30972b310/docs/user/quickstart.rst#json-response-content). You can also find a whole [slew of issues](https://github.com/psf/requests/issues?q=is%3Aissue+simplejson) having being raised around it, including this inconsistent raising of exceptions.

From the discussion [here](https://github.com/psf/requests/issues/3052) and the subsequent removal of `simplejson` [here](https://github.com/psf/requests/pull/3056) (slated for a requests 3.0 release that's still on the horizon), I'm going to make the call for this package that we're not going to entertain this game.
- pymyob doesn't support python2 (i actually plan to drop python 3.5 in the not-too-distant future as well, and f-string everything)
- the performance benefits of `simplejson` over `json` these days aren't huge, and python is now on a yearly release cycle, so the wait for security updates isn't long.
- there are slight nuances in (de)serialisation between `json` and `simplejson`, which I don't want to deal with if they were to crop up.

My suggestions for you:
- drop `simplejson` from your codebase -- probs easier said than done in most situations, but I'ma suggest it anyway.
- add your own try/except block around your pymyob delete calls to catch and handle `simplejson.JSONDecodeError`.
- wait for requests 3 to be released (if/when that happens).
- hassle requests maintainers

**Comment by jpmoral-factory** [comment]
Yeah, I was surprised we were using simplejson. Will probably get rid of it at some point but not in the near future. I'll go with your `ValueError` suggestion.

**Comment by jarekwg** [comment]
Resolved in #60.
